the-collab-lab/tcl-11-smart-shopping-list
# Issue #1 | Connect app to Firebase

## Description

Added three new components: one for referencing the database, one for adding an item to the database, and another one to display items from the database in real-time.

### Referencing database:
Inside Firebase component initialized the dataBase that would allow accessing and updating the firestore.

### Displaying items from the database:
Inside Listener component set up a listener that would monitor the updates in the database and display the changes in real-time. 

### Writing items to the database:
Inside AddItem component wrote a function that would update the database every time a new item is added

## Related Issue

Closes https://github.com/the-collab-lab/tcl-11-smart-shopping-list/issues/1

## Acceptance Criteria


## Type of Changes


|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |


## Testing Steps / QA Criteria

From the terminal, pull down this branch with `git pull origin fc-vh-connecting-to-firestore` and check that branch out with `git checkout c-vh-connecting-to-firestore`
Run `npm install` to install the newly added dependencies locally. Run `npm start` to launch the app.
Click on `Click to get items!` button to display items in the database.
Type the name of the item you'd like to add and click `Add Item` button to add the item to the database. The newly added item should be displayed now.
